pkg/cloud/services/compute/instance.go

```diff
@@ -206,10 +206,20 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
 		})
 	}
 
-	if instanceSpec.Subnet != "" && accessIPv4 == "" {
+	var server *ServerExt
+
+	// Ensure we delete the ports we created if we haven't created the server.
+	defer func() {
```

Comment:
should the defer maybe also be above line 184?

If creating multiple ports and a port fails we may want to cleanup the already created ones too?

Reply:
I can also add a unit test for that.

Reply:
I've squashed this and added a new unit test.